### PR TITLE
Question having multiple accepted answers

### PR DESCRIPTION
**Pull request by andela-ooranagwa**
Fixes this be verifying all the answers to question and return error message if accepted answer already exists.

### CODE REVIEW

File: app/controllers/answers_controller.rb [hunk 1]
```diff
@@ -39,6 +39,7 @@ def destroy
 
   def accept
     return unauthorized_access unless @question.user == current_user
+    return invalid_request('Question has an already accepted answer') if @question.answers.any?(&:accepted)
```
Comment (andela-oakinniranye) [inline app/controllers/answers_controller.rb]: do you think a `before_action :authorize_user, :validate_question_already_has_accepted_answer` works in this case?

Comment (andela-oojewale) [inline app/controllers/answers_controller.rb]: I think that makes sense, the `before_action`.

Comment (andela-ooranagwa) [inline app/controllers/answers_controller.rb]: A `before_action` makes a whole lot of sense. My thoughts however is that having a `before_action` for a **one line** method that is used just by one action is something of an _overkill_

Comment (andela-ooranagwa) [inline app/controllers/answers_controller.rb]: I will be making it into a `before_action` nevertheless _inclusive of the other validation in the action_

**Pull request merged**